specs/schemas/schema-schema.ipldsch

```diff
@@ -425,15 +425,22 @@ type UnionRepresentation_Inline struct {
 
 ## UnionRepresentation_BytePrefix describes a union representation for unions
 ## whose member types are all bytes. It is encoded to a byte array whose
-## first byte is the discriminator and subsequent bytes form the discriminated
+## first bytes are the discriminator and subsequent bytes form the discriminated
 ## type.
 ##
+## Discriminators are represented as hexadecimal strings. There is currently
+## no limitation on their length, other than needing to be at least one byte.
+## Nor is there a requirement that they all be of the same length, although
+## they must all represent unique prefixes.
+##
+## While Schema DSL parsers may allow `0x` prefixes to the hexadecimal strings,
+## they should not be present in the internal representation, only valid
+## hexadecimal strings are allowed.
+##
 ## byteprefix is an invalid representation for any union that contains a type
 ## that does not have a bytes representation.
 ##
-type UnionRepresentation_BytePrefix struct {
-	discriminantTable {TypeName:Int}
-}
+type UnionRepresentation_BytePrefix {String:TypeName}
```

Comment:
We should probably change the name of this to `type UnionRepresentation_BytesPrefix` (with an 's') now, too.  :facepalm: